pandas/core/generic.py

```diff
@@ -4951,9 +4950,7 @@ def pipe(self, func, *args, **kwargs):
 
     _shared_docs['aggregate'] = dedent("""
     Aggregate using one or more operations over the specified axis.
-
```

Comment:
Does this still render the version added directive appropriately?

Reply:
None of the current aggregate functions have version added directives, which is actually what's causing the errors here (three blank lines). In order to add them in the future we'll need to `dedent` the document before its substitution (similar to the `see_also` and `examples` directives), eg:

```
_agg_version_added_doc = dedent("""
Test version added doc
""")

@Substitution(see_also=_agg_see_also_doc,
              examples=_agg_examples_doc,
              versionadded=_agg_version_added_doc,
```

which renders the directive appropriately:

```
################################################################################
############# Docstring (pandas.core.resample.Resampler.aggregate) #############
################################################################################

Aggregate using one or more operations over the specified axis.

Test version added doc

Parameters
```

Reply:
OK makes sense. I think in other places the version added actually gets passed in with a line break preceding the value so this would at least be consistent with other uses